### PR TITLE
Update PNPM to 8.15.5

### LINKED ISSUE #9321: Dependabot::SharedHelpers::HelperSubprocessFailed:  ERR_PNPM_UNSUPPORTED_PLATFORM  Unsupported platform for registry.npmjs.org/@swc/core-darwin-x64/...
Sentry Issue: [DELTAFORCE-YYW](https://github.sentry.io/issues/5059088706/?referrer=github_integration)

```
Dependabot::SharedHelpers::HelperSubprocessFailed:  ERR_PNPM_UNSUPPORTED_PLATFORM  Unsupported platform for registry.npmjs.org/@swc/core-darwin-x64/1.4.8: wanted {"cpu":["x64"],"os":["darwin"],"libc":["any"]} (current: {"os":"linux","cpu":"x64","libc":"glibc"})

This error happened while installing a direct dependency of /home/dependabot/dependabot-updater/repo
Progress: resolved 1, reused 0, downloaded 0, added 0
  common/lib/dependabot/shared_helpers.rb:429:in `run_shell_command'
  dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11193/lib/types/private/methods/call_validation.rb:169:in `bind_call'
  dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11193/lib/types/private/methods/call_validation.rb:169:in `validate_call_skip_block_type'
  dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11193/lib/types/private/methods/call_validation.rb:111:in `block in create_validator_slow_skip_block_type'
  npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb:171:in `run_pnpm_command'
...
(56 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by abdulapopoola**
Addresses a toolVersion bug

Fixes #9321

### CODE REVIEW

File: npm_and_yarn/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ FROM ghcr.io/dependabot/dependabot-updater-core
 ARG COREPACK_VERSION=0.24.0
 
 # Check for updates at https://github.com/pnpm/pnpm/releases
-ARG PNPM_VERSION=8.15.2 
+ARG PNPM_VERSION=8.15.4
```
Comment (yeikel) [inline npm_and_yarn/Dockerfile:7]:
```suggestion
ARG PNPM_VERSION=8.15.5
```

Comment (yeikel) [reply npm_and_yarn/Dockerfile:7]: https://github.com/pnpm/pnpm/releases/tag/v8.15.5

Comment (yeikel) [reply npm_and_yarn/Dockerfile:7]: If you apply this, the title needs to be updated :)

Comment (abdulapopoola) [reply npm_and_yarn/Dockerfile]: Thanks!

**Review by landongrindheim — APPROVED** [review]

**Comment by pboling** [comment]
Bumping the version like this breaks compatibility with every project that is pinned to a specific version of pnpm, which is likely most projects that use pnpm.  There must be a better way.

**Pull request merged**